amulet/deployer.py

```diff
@@ -136,6 +136,15 @@ def add(self, service_name,
         if placement is not None:
             service['to'] = placement
 
+        if constraints is None and 'JUJU_TEST_CONSTRAINTS' in os.environ:
```

Comment:
+1, I understand your point, I just don't want to add code that solves a problem that nobody has. YAGNI and all that. Plus, arguing on github is fun.